### PR TITLE
CASSANDRA-17244: Fix org.apache.cassandra.distributed.test.trackwarnings.TombstoneCountWarningTest.failThresholdSinglePartition

### PR DESCRIPTION
**Pull request by dcapwell**

### CODE REVIEW

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java [hunk 1]
```diff
@@ -68,10 +84,12 @@ public class TombstoneCountWarningTest extends TestBaseImpl
     @BeforeClass
     public static void setupClass() throws IOException
     {
+        logger.info("[test step : @BeforeClass] setupClass");
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java:87]: these tests have very similar logs, so while debugging if we do not annotate test start its very hard to debug

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java [hunk 2]
```diff
@@ -248,7 +274,10 @@ public boolean matches(InetAddress value)
 
         assertWarnAborts(0, 2, 1);
 
+        // when disabled warnings only happen if on the coordinator, and coordinator may not be the one replying
+        // to every query
         enable(false);
+        State.blockFor(CLUSTER.get(1).config().broadcastAddress());
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java:280]: honestly `driverQueryAll(cql)` below could use the same, but for some reason its stable where as this wasn't...

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java [hunk 3]
```diff
@@ -259,7 +288,12 @@ public boolean matches(InetAddress value)
             catch (ReadFailureException e)
             {
                 Assertions.assertThat(e).isNotInstanceOf(TombstoneAbortException.class);
+                Assertions.assertThat(e.failureReasonByEndpoint).isNotEmpty();
+                Assertions.assertThat(e.failureReasonByEndpoint.values())
+                          .as("Non READ_TOO_MANY_TOMBSTONES exists")
+                          .allMatch(RequestFailureReason.READ_TOO_MANY_TOMBSTONES::equals);
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java:294]: this is just for consistency; we do this in the above queries so why not here?

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java [hunk 4]
```diff
@@ -327,4 +361,75 @@ private static ResultSet driverQueryAll(String cql)
     {
         return JAVA_DRIVER_SESSION.execute(new SimpleStatement(cql).setConsistencyLevel(com.datastax.driver.core.ConsistencyLevel.ALL));
     }
+
+    @Shared
+    public static class State
+    {
+        public static volatile InetSocketAddress blockFor = null;
+        public static volatile CompletableFuture<Void> promise = new CompletableFuture<>();
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java]: could be `null`... I just hate nulls ^_^

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java [hunk 5]
```diff
@@ -327,4 +361,78 @@ private static ResultSet driverQueryAll(String cql)
...
+        public static void onFailure(InetAddressAndPort from, RequestFailureReason failureReason, @SuperCall Runnable zuper) throws Exception
+        {
+            State.onFailure(new InetSocketAddress(from.getAddress(), from.getPort()));
+            zuper.run();
+        }
+
+        // make sure to schedule the task rather than running inline...
+        // this is imporant as the read may block on the local version which can get the test to include it rather than
+        // block waiting, so by scheduling we make sure its always fair
+        public static void maybeExecuteImmediately(Runnable task, @This SEPExecutor executor)
```
Comment (maedhroz) [inline test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java:438]: nit: throw an `@SuppressWarnings("unused")` on this method if you're OCD like me about IDEA's compile warnings :D

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java (hunk 4 above)
Comment (maedhroz) [reply test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java]: I guess the benefit of it starting out `null` would be that, if it must be set in `blockFor()`, null would alert us immediately if we had failed to do that? (...or aesthetically, we return it to `null` in `syncAndClear()`, so why not here?)

**Review by maedhroz — APPROVED** [review]

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java (hunk 5 above)
Comment (dcapwell) [reply test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java:438]: you and your warnings!

File: test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java (hunk 4 above)
Comment (dcapwell) [reply test/distributed/org/apache/cassandra/distributed/test/thresholds/TombstoneCountWarningTest.java]: I went with null

**Pull request closed without merging**